source/Cosmos.HAL2/Device.cs

```diff
@@ -1,9 +1,148 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Linq;
 using System.Text;
 
-namespace Cosmos.HAL {
-    public abstract class Device {
+namespace Cosmos.HAL
+{
+    // This is the base class for all hardware devices. Hardware devices 
+    // are defined here as abstracts and overridden in specific implementations
+    // later
+    public abstract class Device
```

Comment:
Not sure if these changes are necessary, as they dont really seem to be fully implemented/used anywhere.